cmd/datasource-syncer/main.go

```diff
@@ -0,0 +1,205 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	grafana "github.com/grafana/grafana-api-golang-client"
+	"golang.org/x/mod/semver"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+)
+
+var (
+	grafanaToken = flag.String("grafana-api-token", "",
+		"grafana-api-token used to access Grafana. Can be created using: https://grafana.com/docs/grafana/latest/administration/service-accounts/#create-a-service-account-in-grafana")
+	grafanaEndpoint    = flag.String("grafana-api-endpoint", "", "grafana-api-endpoint is the endpoint of the Grafana instance that contains the datasources to update.")
+	datasourceUIDList  = flag.String("datasource-uids", "", "datasource-uids is a comma separated list of datasource UIDs to update.")
+	prometheusEndpoint = flag.String("prometheus-endpoint", "",
+		"prometheus-endpoint is the Prometheus server endpoint. This flag updates the Prometheus server URL of the datasource.")
```

Comment:
endpoint. For Managed Service for Prometheus, this will start with 'monitoring.googleapis.com'.

Reply:
no longer used.